File: internal/dto/service.go
package dto

import "time"

type CreateServiceRequest struct {
	Name        string `json:"name" validate:"required,min=3,max=100"`
	DurationMin int    `json:"duration_min" validate:"required,min=1"`
	PriceCents  int    `json:"price_cents" validate:"required,min=1"`
	LocationID  string `json:"location_id" validate:"omitempty"`
	CategoryID  string `json:"category_id" validate:"omitempty"`
	OrderIndex  int    `json:"order_index" validate:"omitempty,min=0"`
}

type UpdateServiceRequest struct {
	Name        string `json:"name" validate:"omitempty,min=3,max=100"`
	DurationMin int    `json:"duration_min" validate:"omitempty,min=1"`
	PriceCents  int    `json:"price_cents" validate:"omitempty,min=1"`
	LocationID  string `json:"location_id" validate:"omitempty"`
	CategoryID  string `json:"category_id" validate:"omitempty"`
	OrderIndex  int    `json:"order_index" validate:"omitempty,min=0"`
}

type ServiceResponse struct {
	ID          string    `json:"id"`
	BusinessID  string    `json:"business_id"`
	LocationID  string    `json:"location_id"`
	Name        string    `json:"name"`
	DurationMin int       `json:"duration_min"`
	PriceCents  int       `json:"price_cents"`
	CategoryID  string    `json:"category_id,omitempty"`
	OrderIndex  int       `json:"order_index"`
	CreatedAt   time.Time `json:"created_at"`
	UpdatedAt   time.Time `json:"updated_at"`
}

// Category DTOs
type CreateCategoryRequest struct {
	Name       string `json:"name" validate:"required,min=3,max=100"`
	OrderIndex int    `json:"order_index" validate:"omitempty,min=0"`
}

type UpdateCategoryRequest struct {
	Name       string `json:"name" validate:"omitempty,min=3,max=100"`
	OrderIndex int    `json:"order_index" validate:"omitempty,min=0"`
}

type CategoryResponse struct {
	ID         string    `json:"id"`
	BusinessID string    `json:"business_id"`
	Name       string    `json:"name"`
	OrderIndex int       `json:"order_index"`
	CreatedAt  time.Time `json:"created_at"`
	UpdatedAt  time.Time `json:"updated_at"`
}

type UpdateOrderRequest struct {
	IDs []string `json:"ids" validate:"required,min=1"`
}

type CategoryWithServices struct {
	CategoryResponse
	Services []ServiceResponse `json:"services"`
}

type BusinessServicesResponse struct {
	Categories []CategoryWithServices `json:"categories"`
	Services   []ServiceResponse      `json:"uncategorized_services,omitempty"`
}

File: frontend/src/types/service.ts
export interface Service {
  id: string
  business_id: string
  location_id?: string
  category_id?: string
  name: string
  duration_min: number
  price_cents: number
  order_index: number
  created_at: string
  updated_at: string
}

export interface CreateServiceRequest {
  name: string
  duration_min: number
  price_cents: number
  location_id?: string
  category_id?: string
  order_index?: number
}

export interface UpdateServiceRequest {
  name?: string
  duration_min?: number
  price_cents?: number
  location_id?: string
  category_id?: string
  order_index?: number
}

export interface ServiceCategory {
  id: string
  business_id: string
  name: string
  order_index: number
  created_at: string
  updated_at: string
}

export interface CreateCategoryRequest {
  name: string
  order_index?: number
}

export interface UpdateCategoryRequest {
  name?: string
  order_index?: number
}

export interface UpdateOrderRequest {
  ids: string[]
}

export interface CategoryWithServices {
  id: string
  business_id: string
  name: string
  order_index: number
  created_at: string
  updated_at: string
  services: Service[]
}

export interface BusinessServicesResponse {
  categories: CategoryWithServices[]
  uncategorized_services?: Service[]
}